### DISCUSSION TITLE
VPN / WireGuard?

### DISCUSSION THREAD

**Discussion by simplenotezy** [homeassistant-apps/app-cloudflared#746] [Q&A] [answered]
Can this addon be used as a VPN as well, so that I can surf from my home IP when abroad? And in that case, how?

I'd like to avoid exposing HA to e.g. WireGuard now that I have Cloudflare Tunnel access

**Comment by brenner-tobias** [comment, marked as answer]
Unfortunately, as far as I am aware this is not possible out-of-the-box since Cloudflare Tunnel does not offer an "Exit-node" option. I found some possible workarounds by fiddling with IP Tables, bot sure though if this works and how stable it is.

For your use-case I suggest to have a look at [Tailscale](https://tailscale.com/kb/1103/exit-nodes) and the respecting [HA add-on](https://github.com/hassio-addons/addon-tailscale). You can use HA as an exit-node and route all traffic of your devices through it.

**Comment by simplenotezy** [comment]
OK thanks @brenner-tobias
